src/OrchardCore.Modules/OrchardCore.Lists/Views/ListPartDetailAdmin.cshtml

```diff
@@ -51,29 +55,45 @@ else
-                                </a>
+                                @if (await AuthorizationService.AuthorizeAsync(User, CommonPermissions.EditContent, contentItem))
+                                {
+                                    <a class="btn btn-sm btn-success" asp-action="Create" asp-controller="Admin" asp-route-id="@contentTypeDefinition.Name" asp-route-area="OrchardCore.Contents" asp-route-ListPart.ContainerId="@Model.ListPart.ContentItem.ContentItemId" asp-route-ListPart.ContentType="@contentTypeDefinition.Name" asp-route-ListPart.EnableOrdering="@Model.EnableOrdering" asp-route-returnUrl="@FullRequestPath">
+                                        @T["Create {0}", contentTypeDefinition.DisplayName]
+                                    </a>
+                                }
                             }
                             else
                             {
+                                var authorizedContentTypeDefinitions = Model.ContainedContentTypeDefinitions
+                                    .Where(contentTypeDefinition =>
+                                        AuthorizationService.AuthorizeAsync(
+                                            User,
+                                            CommonPermissions.EditContent,
+                                            ContentManager.NewAsync(contentTypeDefinition.Name)
+                                        .Result)
+                                    .Result);
+
                                 <a class="btn btn-sm btn-primary" edit-for="@Model.ListPart.ContentItem" asp-route-returnUrl="@FullRequestPath">
                                     @T["Edit {0}", @Model.ListPart.ContentItem.DisplayText]
                                 </a>
-                                <a class="btn btn-sm btn-success dropdown-toggle" href="#" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
-                                    @T["Create"]
-                                </a>
-                                <div class="dropdown-menu" aria-labelledby="dropdownMenuLink">
-                                    @foreach (var containedContentTypeDefinition in Model.ContainedContentTypeDefinitions)
-                                    {
-                                        <a class="dropdown-item" asp-action="Create" asp-controller="Admin" asp-route-id="@containedContentTypeDefinition.Name" asp-route-area="OrchardCore.Contents" asp-route-ListPart.ContainerId="@Model.ListPart.ContentItem.ContentItemId" asp-route-ListPart.ContentType="@containedContentTypeDefinition.Name" asp-route-ListPart.EnableOrdering="@Model.EnableOrdering" asp-route-returnUrl="@FullRequestPath">
-                                            @containedContentTypeDefinition.DisplayName
-                                        </a>
-                                    }
-                                </div>
+                                if(authorizedContentTypeDefinitions.Count() > 0)
```

Comment:
The template previously had a distinction how it generates the HTML for a single content type and more (there is at least a single one every time). You can simplify the whole logic a lot by just building on that but instead of sometimes referencing `Model.ContainedContentTypeDefinitions` always referencing `authorizedContentTypeDefinitions` (that you can produce at the top of the template in the code block).